Search(), get_RoomSize() 구현, 변수 선언

msmf3 opened the pull request:
#Generator.cpp
<pre><code>
void Generator::get_RoomSize(){
	for(int i = 0; i < this.row; i++)
		for(int j = 0; j < this.col; j++)
			if(this->map[i][j] == 1)
				this.roomsize++;
}

int Generator::Search()
{

	/*
	1. Find adjacent position that has to unmakred.
	2. if unexist next position
		1) unmarked recent position.
		2) return lastest position. (back tracking)
	3. if exist next position.
		1) marking this position.
		2) Push position to path and change value from matrix.
		3) move next position
		4) call Search function.
	4. if all path is full fill,
		escape this function.
	*/
  // startingPointx = 0~n
  // startingPointy = 0~m
	int new_x, new_y;

  if ( this->order == this->roomsize )
    return 1;

  for(int side_idx = 0; side_idx < 8; side_idx++){
		//Is_Marked(new_x, new_y, side_idx);
		new_x = this->row + this->side[side_idx][0];
		new_y = this->col + this->side[side_idx][1];
		if ( new_x < 0 || new_x >= this->row ) continue;
		if ( new_y < 0 || new_y >= this->col ) continue;
		if ( this->marked[new_x][new_y] ) continue;		// 이미 방문한 경우 다른 방향으로
		if ( this->map[new_x][new_y] == 0 ) continue;	        // 벽인 경우 다른 방향으로

  	this->map[new_x][new_y] = this->order++;
		path.push_back(this->order);
    this->marked[new_x][new_y] = 1;
    if( Search() ) // 끝까지 도달한 경우.
      return 1;
		this->map[new_x][new_y] = 1;
		path.pop_back();
    this->marked[new_x][new_y] = 0;
  }
  return 0;
}
//run() 에서 Serach를 체크.
if(Search() == 0)
	{
		std::cout<<"False Generator run.\n";
		std::exit(1);
	}
</code></pre>

<hr/>

#Generator.h
<pre><code>
Generator(int row, int col, std::fstream& fop):
	row(row),
	col(col),
	side {{ 0,1 }, { 1,1 }, { 1,0 }, { 1,-1 },
	         { 0,-1 }, { -1,-1 }, { -1,0 }, { -1,1 }}
	{
		this->map = new int*[row];
		for( int i =0; i < row; i++ )
			this->map[i] = new int[col];

		std::cout<<row<<' '<<col<<'\n';
		for(int i = 0; i < row; i++)
		{
			for(int j = 0; j < col; j++)
			{
				fop>>this->map[i][j];
			}
		}

		this->order = 1;
		this->roomsize = 0;
		Setting();
	};

int order;		       // 순서를 매겨주는 변수.
int side[8][2];
int roomsize;                   // 벽을 제외한 방의 크기.

void get_RoomSize();

</code></pre>

LightAllWorld committed:
Upload skeleton code about generator.
- This file is first version skeleton code.

Commit:
1차 뼈대

dnap512 committed:
스켈레톤 1차 수정본

Commit:
인풋 아웃풋 결과 수정

LightAllWorld commented:
if(Search()==0) 부분에서 주석 처리 제대로 안 됨.
주석은 위로 변경해야함.
전체적으로 들여쓰기 맞춰야함.
헤더부분도 생성자 코드를 업로드 해야함

Commit:
1차 recursive_version hidato_puzzle

Commit:
헤더파일의 잘못된 이름 수정 include <Solver3..h> -> include <Solver2.h>

dnap512 committed:
0. recursive 버전 1차 완성본
1. 중간값이 바뀌는 이슈 해결하여 코드 수정.
2. 스켈레톤 코드에서의 설계와 달리 현재 orderOfMat 값이 중간값-1인 경우 인접한 좌표에 현재 순서의 중간값이 있는지 검사를 하지 않고,
현재 노드를 기준으로 검사하는 것으로 변경.
인접한 좌표가 벽인지, 또는 유효한 좌표인지 다시 검사해야 하므로 비효율적이었음.
따라서 인접 좌표를 검사하는 함수 search 삭제

Commit:
0. recursive 버전 1차 완성본에서 잘못쓰여진 헤더파일 수정
